src/enclave/enclave_oe.c

```diff
@@ -303,20 +333,33 @@ static void _copy_shared_memory(const sgxlkl_shared_memory_t* host)
         }
     }
 
-    if (host->env)
+    /* Copy the host's environment variables to enclave memory */
+    char* const* henv = host->env;
+    if (henv)
     {
         size_t henvc = 0;
-        while (host->env[henvc++] != 0)
-            ;
+        while (henv[henvc] != 0)
```

Comment:
`henv` buffer is already accessed before CHECK_OUTSIDE(). If malicious host sets henv pointer to point to secret buffer inside the enlcave, side channel info about which byte within the secret buffer is 0 already leaked.

Reply:
That's correct and expected. Note that the `henv` pointer can't change anymore, but we don't know the size of the memory this points to; we have to search for the first `NULL`. The purpose of this is to import host-controlled information, i.e. the host is welcome to change the contents at any point. All the elements (environment variable settings) are checked again further below.

Reply:
The potential side channel attack is not to change the `henv` pointer after `CHECK_OUTSIDE` line. The attacker's goal is to set `henv` to a secret buffer inside the enclave, then time the NULL search loop to see which byte within the secret buffer is 0.

Reply:
Thanks, you're right, I see what you mean now! I guess that could be closed by adding a check at every iteration of the loop, but I think I'd rather pass the length of the buffer (`henvc`) in as a separate parameter so we can check the location of the whole buffer up front.

Reply:
(Fixed)